### DISCUSSION TITLE
rust-course/about-book

### DISCUSSION THREAD

**Discussion by sunface** [sunface/rust-course#611] [章节评论区]
https://course.rs/about-book.html

**Comment by allandowney** [comment]
很荣幸被列入贡献者名单

学识不足，很惭愧

**Reply by overlords** [reply]
在Rust的天下中，我不允许你再低调了。

**Reply by korman** [reply]
我这学半天学不会的才惭愧，阁下惭愧啥……

**Reply by jieqiyue** [reply]
> 你惭愧个damn

你个damn

**Reply by yuezhenwei** [reply]
你惭愧个damn

**Comment by sunface** [comment]
> @AllanDowney
> 很荣幸被列入贡献者名单
> 
> 学识不足，很惭愧

果然，大佬都是谦虚的 :P

**Reply by ghost** [reply]
老大可否有关如何学习标准库的建议？

**Comment by CaicoLeung** [comment]
前排支持

**Reply by MTDickens** [reply]
借楼，推荐一个 Rust 的所有权可视化，便于初学者理解：https://cel.cs.brown.edu/aquascope/

**Comment by sunface** [comment]
> @CaicoLeung
> 前排支持

感谢支持,  欢迎多多提出宝贵的意见和建议 :)

**Comment by mxismean** [comment]
👍🏻，支持

**Comment by ghost** [comment]
多谢无私奉献。
本人已经有几十年没有编程了。由于疫情，想做点自己喜欢的事情，想到了编程，单纯的兴趣，目标开源。

**Reply by sunface** [reply]
好奇的问一句，当初写什么语言？为何几十年没有编程了🤣

**Reply by ghost** [reply]
C， 汇编，SQL

**Reply by ghost** [reply]
IT不止编程对吧。

**Reply by sunface** [reply]
是的。。为你点赞 😆

**Reply by ghost** [reply]
对不起，刚注意到是ower，
说话有些冒昧了，抱歉。

**Reply by ghost** [reply]
请教sunface，可否有pdf格式的文本？

**Reply by zhaoxm** [reply]
大佬竟然有 9k 个 followers？

**Reply by zhaoxm** [reply]
原来是 ghost...

**Comment by tcqiuyu** [comment]
看了两章，回来前排留名，写的太好了

**Reply by gegekeru** [reply]
放弃一次了，又重新开始了。

**Comment by python8384** [comment]
在生活中碰不到这么伟大的人，在这里碰到了，致敬开源，致敬作者。 👍🏻👍🏻👍🏻

**Reply by sunface** [reply]
这个。。。并没有哈哈，不过能得到你的认可，我也很开心

**Reply by makerjackie** [reply]
谢谢作者!! 真的挺伟大的!! 让我入坑rust 了

**Comment by liuliuliuzy** [comment]
资瓷！

**Comment by caoxiemeihao** [comment]
Starts with [2022-04-19 17:15:49]

**Reply by Rustln** [reply]
Come on 👍

**Comment by tieshulin** [comment]
我来了，感谢作者，愿我们都能成为开源大神。

**Reply by sunface** [reply]
加油～～

**Comment by ManonLoki** [comment]
说实话 这套圣经我翻了半年…… 这已经是第三次从头过了 每次都有新的收获 感谢

**Reply by sunface** [reply]
半年前。。。是在翻 TODO 章节嘛？😆

**Comment by fengli0** [comment]
感谢各位大佬的无私奉献，为你们点赞👍🏻👍🏻👍🏻

**Comment by shilohooo** [comment]
看Rust程序设计语言中文版学习到枚举，然后发现这个网站，古德古德

**Reply by csx12588** [reply]
你发现宝藏了，不过这半年了，学的如何了

**Reply by shilohooo** [reply]
学到所有权的时候懵逼了，弃坑了....

**Reply by gqingc** [reply]
现在怎么样了？(●'◡'●)

**Comment by zero-edge-code** [comment]
Java/Golang码农，直呼好家伙。越来越卷的后端，有必要看看此书。贡献者们和作者辛苦了。

**Comment by zhaojuncw** [comment]
太棒了捏，太棒了捏，太棒了了捏

**Comment by pphboy** [comment]
送给你小心心，送你花一朵，你在我生命中太多的感动

**Reply by pphboy** [reply]
看完官方文档后，再来看圣经，希望有朝一日能做出点什么有用的东西出来

**Comment by bigger-boss** [comment]
js转golang，现在转rust，初心是编写出尽量高效的程序。看好rust，看好作者！

**Comment by CListery** [comment]
膜拜大佬

**Comment by jyutjyu** [comment]
很好的教程

**Comment by james-wangx** [comment]
发现宝藏！！！

**Comment by moruifei** [comment]
前端也开始卷rust了，三番五次从入门到放弃。找了很久的学习资料终于有幸遇到了作者等开源大神，希望能有所收获。先学为敬！

**Comment by Iurmy** [comment]
主保佑你，R门

**Comment by ZhongFuze** [comment]
冲！！！很开心在入门的时候检索到这份资料

**Comment by microhardsmith** [comment]
支持支持，一起加油

**Comment by Neisseri** [comment]
回来感谢无私的作者与珍贵的开源精神（￣▽￣）

**Comment by yuhao00** [comment]
搜索功能似乎不支持中文？

**Reply by sunface** [reply]
是的～

**Comment by qincasin** [comment]
支持 支持

**Comment by lcy101u** [comment]
來留個言

**Comment by cheekhan** [comment]
马上开始学习，学不会就退星🤤

**Comment by iamwilliamli** [comment]
加油！

**Comment by ghost** [comment]
Cheer up

**Comment by jtyoui** [comment]
牛逼up主，没文化。只能这样表达感情

**Comment by DA-MO-WANG** [comment]
you are godd !

**Comment by VicNine** [comment]
感谢作者

**Comment by Mintnoii** [comment]
作为 Rust 初学者，只是做了一些文档校正与排版美化的工作，很荣幸被列入贡献者名单
大家一起加油，做 “困难” 但正确的事~😀

**Comment by Cyc1ops** [comment]
感谢作者和社区contributor们

**Comment by zk-hi** [comment]
感谢，已星

**Comment by ZLD1819858** [comment]
刚从评论区看到这个宝藏连接过来，先star了！

**Comment by xiaobaistart** [comment]


**Reply by xiaobaistart** [reply]
我给在座的各位丢人了，看来真的没入门就要放弃这个了。还是换一种编程语言学习吧我

**Reply by Iurmy** [reply]
还行吧，这也是个学习过程，win下配置环境变量你早晚也是要知道的

**Reply by ghost** [reply]
> 我并不认为这本书是小白能看懂的书，安装部分都看不懂...很惭愧，什么都是写的理所当然的，比如什么在终端执行以下命令，其位于：%Visual Studio 安装位置%\VC\Tools\MSVC%version%\bin\Hostx64\x64（请自行替换其中的 %Visual Studio 安装位置%、%version% 字段）下。 完全看不懂。
据你自我描述，你确实太白了，几乎没有计算机最基本的知识。
建议你别学了，学rust，至少要通晓一门编程语言。

**Comment by NashAiomos** [comment]
好书

**Comment by TDOPR** [comment]
感觉作者已经将 各种语言融会贯通   感谢辛苦付出  为你们点赞

**Comment by xichaoli** [comment]
致敬作者们！

**Comment by xc2f** [comment]
我学习Rust从入门到放弃已经两次了
希望这是最后一次
哦，不
希望没有下一次

**Reply by ZLD1819858** [reply]
好奇你上次到哪里放弃了，我这第一次看生命周期....有种要放弃的感觉

**Reply by ghost** [reply]
生命周期应该不难理解，函数返回值与参数的引用有关才有生命周期，其它情况该干什么干什么。

**Reply by xc2f** [reply]
> 好奇你上次到哪里放弃了，我这第一次看生命周期....有种要放弃的感觉

是的，几乎都是看到官方文档的[泛型、Trait 和生命周期](https://kaisery.github.io/trpl-zh-cn/ch10-00-generics.html#%E6%B3%9B%E5%9E%8Btrait-%E5%92%8C%E7%94%9F%E5%91%BD%E5%91%A8%E6%9C%9F)这里

**Reply by ZLD1819858** [reply]
> > 好奇你上次到哪里放弃了，我这第一次看生命周期....有种要放弃的感觉
> 
> 是的，几乎都是看到官方文档的[泛型、Trait 和生命周期](https://kaisery.github.io/trpl-zh-cn/ch10-00-generics.html#%E6%B3%9B%E5%9E%8Btrait-%E5%92%8C%E7%94%9F%E5%91%BD%E5%91%A8%E6%9C%9F)这里

一鼓作气！这次一冲到底！

**Comment by deneschen** [comment]
Just Read the Fucking Doc!为作者点赞！👍👍👍

**Comment by stefenMiao** [comment]
感谢大佬们😍

**Comment by ZLD1819858** [comment]
知识点干货满满，前期作为入门学习的好资料，后期也是一本好工具书，最关键的是每章节下都有评论区，这点太赞了！！

**Comment by lightnine** [comment]
支持

**Comment by csx12588** [comment]
学了大概两周了，刚把基础入门让你部分学完，特此做个留念

**Comment by hongjian2015100** [comment]
作者功德无量

**Comment by WeixinX** [comment]
打开学习！

**Comment by davyfy** [comment]
有趣的坛子

**Comment by daodaolee** [comment]
小前端入门程序员，对rust不是很了解，但是很喜欢开源世界。看完了创作感悟后，有被深深感动到，决定成为一名rust学习者（PS：如果有一天成为了开源者，希望也会有作者那样的胸襟和态度）

**Comment by ghost** [comment]
加油加油发自我的 iPhone在 2022年12月9日，15:49，道道里 ***@***.***> 写道：﻿
小前端入门程序员，对rust不是很了解，但是很喜欢开源世界。看完了创作感悟后，有被深深感动到，决定成为一名rust学习者（PS：如果有一天成为了开源者，希望也会有作者那样的胸襟和态度）

**Comment by Hoteryoung** [comment]
Rust小白，有Python经验，有什么能为社区做的么，想边学边贡献呢？

**Comment by just4id** [comment]
全民学Rust吧，再造一个开源盛世，如何？

**Reply by sunface** [reply]
我挺你

**Comment by ttxs25830** [comment]
我想了想，还是指出来吧：文中说自己是“开源”书其实是混淆概念。

开源指的是开放源代码，而实际上书不可能做到不开放“源代码”，对吧？这里开源明显实际上指的是免费。

如果想用一点高级的词汇，我建议使用“社区驱动”。开源用来形容一本书实在不太对

PS.开源和免费是并列关系，不是包含关系。理论上你完全可以设计一个协议仅允许付费获取源代码和二进制，这样就是开源但并不免费的，不过就像付费程序一般都会带有防盗版模块一样，几乎没有人这么做。

**Reply by sunface** [reply]
不知道你为何纠结于这个？放在github上免费阅读的书，不叫开源叫啥呢？或者说就不能叫开源了？你没听说过开源电子书的说法吗？

**Reply by sunface** [reply]
我还必须得跟您较真下了，markdown 都开源的，不叫开源叫什么？难道普通的电子书你还能获取可以直接使用的原格式文档？

如果都这样了，你还是坚持说是，就是不算开源，OK，算你赢，问题不大哈。

**Reply by ttxs25830** [reply]
> 我还必须得跟您较真下了，markdown 都开源的，不叫开源叫什么？难道普通的电子书你还能获取可以直接使用的原格式文档？
> 
> 如果都这样了，你还是坚持说是，就是不算开源，OK，算你赢，问题不大哈。

书的“源”又不是格式，而是内容。只要文字和配图被发布了，那就应该算作发布了“源”。就像一个项目只要遵守一个开放的语言标准（在这里对应着中文本身）不会因为使用了闭源编译器（PDF）而被定性为闭源项目，也不应该因为使用开源编译器就被认为是开源项目。
也许在这里探讨这个问题不太合适，那我道歉。

**Reply by ttxs25830** [reply]
> 不知道你为何纠结于这个？放在github上免费阅读的书，不叫开源叫啥呢？或者说就不能叫开源了？你没听说过开源电子书的说法吗？

看见过，但是我始终觉得这个说法不太对。

**Reply by idrey** [reply]
你想说的难道是知识共享Creative Commons吗

**Comment by suyanan** [comment]
支持！有PDF版本吗？多谢

**Comment by lSpikeSpiegel** [comment]
向大佬们致敬~

**Comment by puxiao** [comment]
非某某教的人，看到“圣经”这个词会略感不舒服，但这一丝没有影响我对该书的敬佩之心。
用爱发电的程序员值得点赞赞赞~
作为一名前端，在立贴为证，今年要真的开始学习 Rust，而不仅仅停留在计划中！
就认定本书了。

**Reply by sunface** [reply]
哎～～当初开始写的时候，目标就是随便试试，所以起了一个自我感觉牛逼的名字，没想到。。。哈哈，所以忽略这个名字吧

**Comment by rzx007** [comment]
十分感谢，多谢无私奉献👍🏻👍🏻👍🏻

**Comment by zcg** [comment]
# 2023/3/24 打卡
我要当铁头娃

**Comment by isrun** [comment]
支持

**Comment by cserwen** [comment]
> 你们用指尖绘制的星空，那里繁星点点，每一颗都在鼓励着怀揣着开源梦想的程序员披荆斩棘、不断前行，不夸张的说，没有你们，开源世界就没有星光，自然也就不会有今天的开源盛世。

看到这里内心很感动，看的出来，作者是真的想要教学，非常感谢每一位贡献者！！！

**Comment by himcsam** [comment]
感谢作者的开源！已经读完基础部分了，帮助很大
（已经在力扣上用rust刷上了QWQ 
作者写的超棒的！加油加油

**Comment by qehshub** [comment]
感谢作者无私奉献！未看先点个赞，不过根据过往经验必是本好书。希望一起壮大中国开源社群

**Comment by iamnoobXD** [comment]
速来加入rust大军行列！！！！！

**Comment by zhaohang15639535703** [comment]
感谢作者们，写得真不错

**Comment by MySetsuna** [comment]
支持支持！为什么时间停留在22年？？

**Comment by realHakeen** [comment]
膜拜大佬们！

**Comment by emojiiii** [comment]
这是我第五次进来学习了...

**Reply by sunface** [reply]
所以。。这本书还是没有教会大家 ：（

**Comment by xflcx1991** [comment]
看了目录和头两节就知道诚意满满，是否考虑出书呢。

**Comment by steven1041** [comment]
早发现这里，我就不会看《rust程序设计》了，看那个文档，分分钟一万匹草泥马在心中奔腾。
在这里，轻松娱快的花一个周末看完了基础部分

**Reply by sunface** [reply]
不太流畅的文字表达，确实废脑细胞 :D

**Comment by litingyes** [comment]
打卡

**Comment by kuncle** [comment]
很不错，已cx

**Comment by reymont** [comment]
很好的阅读体验，总是能够在这里找到答案。感谢作者的贡献。

**Comment by kongiok** [comment]
很榮幸能看的到這麼棒的教程。
謝謝，你們太棒了。

**Comment by 3517277** [comment]
如果有PDF版本就好了

**Reply by sunface** [reply]
rust-course 使用 mdbook 构建电子书，但是它对 pdf 的支持还处于起步阶段，无法生成目录..

**Reply by xflcx1991** [reply]
@3517277 @sunface 
我搜了下，看到3种方案
1. https://www.cnblogs.com/liuchao719/p/generate_bookmark_for_mdbook_pdf.html
2. https://github.com/me1ting/trpl-zh-cn-pdf
3. https://github.com/HollowMan6/mdbook-pdf/blob/main/README_CN.md

第3个方案应该最好，最能融入 mdbook

**Reply by sunface** [reply]
@xflcx1991 第三种试过了 = ， =，pdf-outline 那个插件会报错，另外两种我看看哈

**Reply by 3517277** [reply]
这书写得太好 ，是我看到的RUST中最好的。我怕后面会发生变故，看不了。但我试了3种方法，好像都不理想呀。都是遇到了报错。

**Reply by 3517277** [reply]
每一个装PY，总提示插件问题，无法使用。第二个，作者没有提供EXE文件，第三个，嗯，要LINUX，虽然有子系统，但最后还是报错，不知道是不是子系统原因。

**Reply by 3517277** [reply]
第三个方案，还是不行，用了代理也不行，cargo install mdbook-pdf 总有一个错误，error: failed to run custom build command for mdbook-pdf-headless_chrome v0.1.10 十分头痛。

**Reply by sunface** [reply]
确实难搞～～ ：P

**Reply by sunface** [reply]
倒是不会有变故，记住网址就好，哈哈。。

**Comment by shoggothforever** [comment]
有这么好的学习资源，再也禁不住ferris的诱惑了，学习rust的第0天！

**Comment by A-floweR-y** [comment]
我还没看，就光看《关于本书》就非常敬佩作者和贡献者们的付出。可惜我只能在github上点一颗星。

**Comment by borneygit** [comment]
搞Android的，对rust很感兴趣，该书质量绝对上乘，值得推荐

**Comment by j4go** [comment]
感谢开源，感谢伟大的组织者。Mark 23.08.28

**Comment by Yunnathai** [comment]
定个小目标半年内掌握rust,  2023.09.01

**Comment by livvyguo** [comment]
棒

**Comment by Wangdaxia666** [comment]
hello

**Comment by huanghaox** [comment]
每一遍Rust圣经

**Comment by floiges** [comment]
大佬，必须支持

**Comment by bennyhuo** [comment]
支持~！

**Comment by hongruiwnag** [comment]
大佬，这锈书有pdf版本的么，球球

**Comment by ninakitty** [comment]
> [!IMPORTANT]
> 非常感谢作者的贡献！！！

**Comment by TheCutestCat** [comment]
支持，这个给我很大的帮助👍

**Reply by TomIsYourName** [reply]
感谢作者的付出!

**Comment by yjshi2015** [comment]
早就默默关注rust很久了，今天下决心开始学起！！！

**Comment by TonyJayChen** [comment]
大哥大姐，你们真帅！

**Comment by fiveeth** [comment]
谢谢你们！

**Comment by kimuxer** [comment]
希望能出纸质版的书

**Comment by mzdk100** [comment]
学习rust四个字总结“先苦后甜”，虽然rust的大门不好进去，但一但跨越过去会让你特别自信，当其他程序员加班熬夜找bug的时候，你已经在享受生活了。

**Comment by mouxue23** [comment]
终于开始学习rust啦，mark一下

**Comment by mmfp** [comment]
感谢作者

**Comment by coderew** [comment]
感谢作者分享

**Comment by zhangping99** [comment]
我正在将rust项目迁移到LoongArch架构上，我想知道如何辨别rust的哪些库和架构相关？有没有大神可以探讨一下？

**Comment by sharkrem** [comment]
还没开始学习Rust, 但是佩服作者贡献精神！

**Comment by mzdk100** [comment]
赶快学起来，现在我已经离不开rust了。

**Comment by Cinea4678** [comment]
> 如果以找工作为标准，那国内环境确实还不好  

初看不识其中意，再看已是话中人

**Comment by KUN-YKC** [comment]
入坑，学习新知识，mark！！

**Comment by mzdk100** [comment]
恭喜

**Comment by westYY123** [comment]
EluvK，我滴超人

**Comment by buchenglei** [comment]
兄弟们, 我又来入门了 😄

**Comment by Doub1e007** [comment]
从操作系统训练营第一次接触Rust 相遇于此 致敬开源 致敬作者

**Comment by fengmao** [comment]
打个小广告，我个人全栈开发的apk下载站点 https://liusha.app ，欢迎大家体验

**Comment by wafeishushu** [comment]
真的牛批

**Comment by gh503** [comment]
开始学习，感谢大家！

**Comment by undee** [comment]
使用 mdbook build 创建的 html ,没有目录，记得以前的版本有目录

**Comment by mzdk100** [comment]
droid-wrap是一个针对安卓API封装的高级库，使用时非常简洁干净。这是我业余时间维护的库，希望大家可以给个star。
https://gitcode.net/mzdk100/droid-wrap.git

**Comment by cpython666** [comment]
Python斗罗学Rust第一天，在这一章不仅对于本书有了了解，也感叹作者的文采斐然👍。因为有梦！我们相遇。⛽️

**Comment by yaolideveloper** [comment]
你好，这本书哪里有的买啊？

**Reply by mzdk100** [reply]
直接在这里阅读呀，免费的东西
[Rust语言圣经](https://course.rs/about-book.html)

**Comment by mzdk100** [comment]
还有两本书也很不错：
[Rust入门秘籍](https://rust-book.junmajinlong.com/)
[Rust 程序设计语言 简体中文版](https://kaisery.github.io/trpl-zh-cn)

**Comment by KipJayChou** [comment]
# 你们简直就是圣人

**Comment by lcy101u** [comment]
第三次進入RUST 搞得像在追姑娘一樣

**Comment by mzdk100** [comment]
入门是比较难。

**Comment by jieqiyue** [comment]
我是你樾哥，rust第一天，打卡

**Comment by WilliamColton** [comment]
我之前居然没有点star！实在惭愧，已补上

**Comment by mzdk100** [comment]
发个招聘广告，非诚勿扰：
Rust开发工程师
【岗位职责】
1、负责元宇宙类应用的核心功能开发和技术攻关；
2、解决产品在实现过程中的技术难题，应用前沿技术提升产品的核心竞争力；
3、负责项目的性能优化，保证应用在低端手机的质量；

【岗位要求】
1、统招本科及以上学历（熟悉bevy游戏引擎者优先），具备相关编程开发经验，精通rust编程语言，热爱编程，基础扎实，对数据结构、算法有一定了解，有良好的编程习惯；
2、具备独立工作能力和解决问题的能力、善于总结分享，喜欢动手实践；
3、善于沟通，乐于合作，熟悉软件开发流程；
4、执行力强，对性能敏感，对代码和应用体验精益求精
5、良好的沟通和协作，能跟其他部门同事很好的沟通和合作

期待您的加入，一起共创辉煌
工作地点：北京
简历投递：530562053@qq.com

**Comment by woxin123** [comment]
这本书写的并不好，前面的很多例子会用到后面将的语法，导致一头雾水，很难理解。

**Comment by a981008** [comment]
反复入门 rust

**Comment by mzdk100** [comment]
从入门到踹门而出吗？不过我没这么惨。

**Comment by labyte** [comment]
先不管，第一眼看到这个页面就先去克隆下来，写自己的笔记了，章节和页面目录很棒

**Comment by wrb2012** [comment]
为什么这个网站无法用中文关键词搜索

**Comment by taywang** [comment]
好耶

**Comment by Moonsimon** [comment]
支持！！

**Comment by labyte** [comment]
大佬，我用你这个项目作为笔记的模板，有两个问题请教下啊
1、默认所有的章节都是展开状态怎么设置
2、当有多个层级的时候，现在是点击一个会同时关闭其他小节，怎么改为不自动关闭其他小节呢，自动关闭的化，导致页面跳动，晃得很

**Comment by zengxiong1995** [comment]
学起来。干好几年java了，学这个有压力吗？

**Comment by Chesszyh** [comment]
必须给star了，开源精神万岁！感谢贡献者们

**Comment by Dubley123** [comment]
路人，纯好奇，这本书和英文版的rust-lang/book的关系是什么？在阅读两本书的过程中发现大量重叠内容？如果是借鉴了英文版的内容请问在哪里有援引的标记呢？如果是授权/合作的话在哪里提及许可声明呢？还是直接搬下来没有声明source呢？

**Reply by sunface** [reply]
**首先**，你说的对，本书在内容上大量借鉴了英文原版，毕竟它是标杆，太优秀了，但也不是照搬，毕竟这样做就失去了自己的灵魂。
**其次**，在 Readme.md 里有讲到对英文书的引用，为了不破坏内容的整体性，没有在具体的章节里体现(如果你仔细看，就会发现跟英文书有很多很多的不同，整体理念就很不同)。
**最后**，英文书是 MIT + Apache 协议，本来我可以不用回应你，毕竟还挺浪费时间，但是出于对版权的尊重，还是声明下吧。

PS：我花了这么多精力，不赚广告、不搞社区、不出书，一分钱都没拿到，还被质疑是不是侵权，有点冤枉:D 你应该质疑把英文版直接翻译后拿去出版的大佬们 :P

**Comment by 105u9** [comment]
书很不错，人机很喜欢。
The book is really good, and I really like it.

**Comment by master-Endorsie** [comment]
#### 入学了wwwww

**Comment by Zero-C-Star** [comment]
Once you go Rust, you never go back!

**Comment by vincent-sjh** [comment]
正在用Rust写一个组件化内核OS，祝我好运

**Comment by mekefly** [comment]
感觉学会了，3年前，js都不怎么会写的时候开始学，然后最终放弃了，最近一个月又尝试学习，发现好像没多难，现在感觉是会了，轻松写多线程

**Comment by Appleuiy** [comment]
萌新刚开始学rust，希望自己有一天也能像各位大佬一样为开源贡献一份力

**Comment by xiaocan165-spec** [comment]
感谢开源

**Comment by felixtensor** [comment]
支持！

**Comment by CCChisato** [comment]
等我学成了一定会仔细审核ai代码，到处给大伙的小项目点star+改代码

**Comment by wtn98498** [comment]
AI时代了，想借着学rust补补编程知识，现在是码盲啊:D

**Reply by sunface** [reply]
> AI时代了，想借着学rust补补编程知识，现在是码盲啊:D

好眼光。加油，AI 时代，编程反而是更值得投资的通用能力。
